search-dsls/src/commonMain/kotlin/com/jillesvangurp/searchdsls/querydsl/join-queries.kt

```diff
@@ -0,0 +1,103 @@
+@file:Suppress("unused", "UnusedReceiverParameter")
+
+package com.jillesvangurp.searchdsls.querydsl
+
+import com.jillesvangurp.jsondsl.JsonDsl
+import com.jillesvangurp.jsondsl.json
+
+enum class ScoreMode {
+    avg,min,max,none,sum
+}
+
+@SearchDSLMarker
+class NestedQuery : ESQuery(name = "nested") {
+    var path: String by queryDetails.property()
+    var query: ESQuery by queryDetails.esQueryProperty()
+    var scoreMode: ScoreMode by queryDetails.property()
+    var ignoreUnmapped: Boolean by queryDetails.property()
+}
+
+fun QueryClauses.nested(block: NestedQuery.() -> Unit): NestedQuery {
+    val q = NestedQuery()
+    block.invoke(q)
```

Comment:
minor - direct call can be used here and below

```suggestion
    block(q)
```

Reply:
I used `invoke()` as it seems like most of the project is using it over direct call and I wanted to keep with convention of the project. But I am also happy to change to direct call